x-pack/platform/plugins/private/monitoring/server/es_client/instantiate_client.test.ts

```diff
@@ -29,85 +34,80 @@ const serverWithUrl = {
-      instantiateClient(serverWithUrl.monitoring.ui.elasticsearch, log, createClient);
+      instantiateClient(serverWithUrl.monitoring.ui.elasticsearch, log, elasticsearchStart);
 
       expect(log.info.mock.calls[0]).toEqual(['config sourced from: monitoring cluster']);
     });
   });
 
   describe('Custom Headers Configuration', () => {
-    it('Does not add xpack.monitoring.elasticsearch.customHeaders if connected to production cluster', () => {
-      instantiateClient(server.monitoring.ui.elasticsearch, log, createClient);
+    it('Does not create a new client if connected to production cluster', () => {
+      instantiateClient(server.monitoring.ui.elasticsearch, log, elasticsearchStart);
 
-      const createClusterCall = createClient.mock.calls[0];
-      expect(createClient).toHaveBeenCalledTimes(1);
-      expect(createClusterCall[0]).toBe('monitoring');
-      expect(createClusterCall[1].customHeaders).toEqual(undefined);
+      expect(elasticsearchStart.createClient).toHaveBeenCalledTimes(0);
     });
 
     it('Adds xpack.monitoring.elasticsearch.customHeaders if connected to monitoring cluster', () => {
-      instantiateClient(serverWithUrl.monitoring.ui.elasticsearch, log, createClient);
+      instantiateClient(serverWithUrl.monitoring.ui.elasticsearch, log, elasticsearchStart);
 
-      const createClusterCall = createClient.mock.calls[0];
+      const createClusterCall = elasticsearchStart.createClient.mock.calls[0];
 
-      expect(createClient).toHaveBeenCalledTimes(1);
+      expect(elasticsearchStart.createClient).toHaveBeenCalledTimes(1);
       expect(createClusterCall[0]).toBe('monitoring');
-      expect(createClusterCall[1].customHeaders).toEqual({
+      expect(createClusterCall[1]?.customHeaders).toEqual({
         'x-custom-headers-test': 'connection-monitoring',
       });
     });
   });
 
   describe('Use a connection to production cluster', () => {
     it('exposes an authenticated client using production host settings', () => {
-      instantiateClient(server.monitoring.ui.elasticsearch, log, createClient);
+      instantiateClient(server.monitoring.ui.elasticsearch, log, elasticsearchStart);
 
-      const createClusterCall = createClient.mock.calls[0];
-      const createClientOptions = createClusterCall[1];
-
-      expect(createClient).toHaveBeenCalledTimes(1);
-      expect(createClusterCall[0]).toBe('monitoring');
-      expect(createClientOptions.hosts).toEqual(undefined);
+      expect(elasticsearchStart.createClient).toHaveBeenCalledTimes(0);
     });
   });
 
   describe('Use a connection to monitoring cluster', () => {
     it('exposes an authenticated client using monitoring host settings', () => {
-      instantiateClient(serverWithUrl.monitoring.ui.elasticsearch, log, createClient);
-      const createClusterCall = createClient.mock.calls[0];
+      instantiateClient(serverWithUrl.monitoring.ui.elasticsearch, log, elasticsearchStart);
```

Comment:
Similar to the above, since this should return the result `createClient`, we could expect it to be the same that the mocked call returns.